micro_dl/utils/masks.py

```diff
@@ -94,3 +95,65 @@ def create_unimodal_mask(input_image, str_elem_size=3):
     thr_image = binary_opening(input_image > thr, str_elem)
     mask = binary_erosion(thr_image, str_elem)
     return mask
+
+
+def get_unet_border_weight_map(annotation, w0=10, sigma=5):
+    """
+    Generate the weight map for borders as specified in the UNet paper for an annotation.
+    Parameters
+    ----------
+    annotation: array-like
+        A 2D array of shape (image_height, image_width) contains annotation with each class labeled as an integer.
+
+    Returns
+    -------
+    array-like
+        A 2D array of shape (image_height, image_width)
+
+    TODO: Calculate boundaries directly and calculate distance from boundary of cells to another
+    TODO: The below method only works for UNet Segmentation only
+    """
+
+    assert annotation.dtype == np.uint8, "datatype expected uint8, it is {}".format(annotation.dtype)
+    # class balance weights w_c(x)
+    unique_values = np.unique(annotation).tolist()
+    weight_map = [0] * len(unique_values)
+    for index, unique_value in enumerate(unique_values):
+        mask = np.zeros((annotation.shape[0], annotation.shape[1]), dtype=np.float64)
+        mask[mask == unique_value] = 1
+        if mask.sum() != 0:
```

Comment:
Not sure if u need an if-else here as unique_values are the unique values in the annotation and the count will never be zero..